### ISSUE TITLE
Dartdoc fails with new function type syntax

### ISSUE THREAD

**Issue by greglittlefield-wf** [dart-lang/dartdoc#1483]
Input (`lib/test_dartdoc.dart`):
```dart
String Function(int, bool) compute;
```

Output
```
$ pub run dartdoc
Generating documentation for 'test_dartdoc' into /Users/greglittlefield/Development/test_dartdoc/doc/api/

parsing lib/test_dartdoc.dart...
parsed 1 file in 2.0 seconds

Generation failed: Unknown type GenericFunctionTypeElementImpl
package:dartdoc/src/model.dart 1991                                             new ModelElement.from
package:dartdoc/src/model.dart 4076                                             new TopLevelVariable
package:dartdoc/src/model.dart 1979                                             new ModelElement.from
package:dartdoc/src/model.dart 1602                                             Library._getVariables.<fn>
dart:core                                                                       Iterable.toList
package:dartdoc/src/model.dart 1603                                             Library._getVariables
package:dartdoc/src/model.dart 1338                                             Library.constants
package:dartdoc/src/model.dart 1640                                             Library.modelElementsMap
package:dartdoc/src/model.dart 1666                                             Library.allModelElements
package:dartdoc/src/model.dart 3838                                             Package.allModelElements.<fn>
dart:core                                                                       List.forEach
package:dartdoc/src/model.dart 3837                                             Package.allModelElements
package:dartdoc/src/model.dart 3847                                             Package.allCanonicalModelElements
package:dartdoc/dartdoc.dart 181                                                DartDoc.generateDocs
../../.pub-cache/hosted/pub.dartlang.org/dartdoc-0.13.0+1/bin/dartdoc.dart 200  main.<fn>
package:stack_trace                                                             Chain.capture
../../.pub-cache/hosted/pub.dartlang.org/dartdoc-0.13.0+1/bin/dartdoc.dart 199  main
===== asynchronous gap ===========================
dart:async                                                                      _Completer.completeError
package:dartdoc/dartdoc.dart 216                                                DartDoc.generateDocs
===== asynchronous gap ===========================
dart:async                                                                      new Future.microtask
package:dartdoc/dartdoc.dart 128                                                DartDoc.generateDocs
../../.pub-cache/hosted/pub.dartlang.org/dartdoc-0.13.0+1/bin/dartdoc.dart 200  main.<fn>
===== asynchronous gap ===========================
dart:async                                                                      new Future.microtask
../../.pub-cache/hosted/pub.dartlang.org/dartdoc-0.13.0+1/bin/dartdoc.dart 199  main.<fn>
package:stack_trace                                                             Chain.capture
../../.pub-cache/hosted/pub.dartlang.org/dartdoc-0.13.0+1/bin/dartdoc.dart 199  main
```

Versions:
- Dart SDK 1.24.2
- dartdoc 0.13.0+1
- analyzer 0.30.0+2

**Comment by devoncarew** [comment]
@jcollins-g @bwilkerson; we'll need to rev dartdoc to use the last published alpha version of package:analyzer, or publish a new stable version of analyzer w/ the function type syntax fixes.

**Comment by jcollins-g** [comment]
Fixed in #1506
